src/codecs/avif/decoder.rs

```diff
@@ -0,0 +1,160 @@
+//! Decoding of AVIF images.
+///
+/// The [AVIF] specification defines an image derivative of the AV1 bitstream, an open video codec.
+///
+/// [AVIF]: https://aomediacodec.github.io/av1-avif/
+
+use std::convert::TryFrom;
+use std::io::{self, Cursor, Read};
+use std::marker::PhantomData;
+use std::mem;
+
+use crate::{ColorType, ImageResult, ImageDecoder, ImageFormat, ImageError};
+use crate::error::DecodingError;
+
+use mp4parse::read_avif;
+use dav1d::{PlanarImageComponent, PixelLayout};
+use dcv_color_primitives as dcp;
+
+impl From<mp4parse::Error> for ImageError {
+    fn from(err: mp4parse::Error) -> ImageError {
+        ImageError::Decoding(DecodingError::new(ImageFormat::Avif.into(), err))
+    }
+}
+
+impl From<dav1d::Error> for ImageError {
+    fn from(err: dav1d::Error) -> ImageError {
+        ImageError::Decoding(DecodingError::new(ImageFormat::Avif.into(), err))
+    }
+}
+
+impl From<dcp::ErrorKind> for ImageError {
+    fn from(err: dcp::ErrorKind) -> ImageError {
+        ImageError::Decoding(DecodingError::new(ImageFormat::Avif.into(), err))
+    }
+}
```

Comment:
This turns all of those crates into public dependencies. We usually don't want that to happen for any image format whose encoder/decoder hasn't reached a stable version yet.

Reply:
Hm, one of the CI jobs should have linted this: https://github.com/image-rs/image/runs/1772778509
I wonder why not. It might have to do with this not being a default feature.

Reply:
Hmm, should I do the conversion manually then, instead of using `?` in the code?  I previously had a quite heavy `.map_err()` doing the same conversion, but that made the code bothersome to read.

I could go back to the `.map_err()` way, but using a helper conversion function, would that be ok?

Reply:
That’s what I did, with the new `error_map()` function which goes in the `.map_err()` to replace this `From` impl.